### DISCUSSION TITLE
Opaque textures after switch to VSG 1.1.4

### DISCUSSION THREAD

**Discussion by avilkinnick** [vsg-dev/VulkanSceneGraph#1210] [Q&A] [answered]
Hello, I was loading my textures with stbi like this:
```
const vsg::Path texture_file = vsg::findFile(texture_path, options);
if (vsg::fileExtension(texture_file) == ".bmp")
{
    stbi_set_flip_vertically_on_load(1);
}
else
{
    stbi_set_flip_vertically_on_load(0);
}

int width, height, channels;
stbi_uc* pixels = stbi_load(texture_file.string().c_str(), &width, &height, &channels, STBI_rgb_alpha);
texture_data = vsg::ubvec4Array2D::create(width, height, reinterpret_cast<vsg::ubvec4*>(pixels), vsg::Data::Properties{VK_FORMAT_R8G8B8A8_UNORM});
```
and everything worked fine, but when i switched to VSG 1.1.4, textures with alpha channel that had trasparent parts were shown with black or white rectangle like thay hadn't alpha channel

**Comment by robertosfield** [comment, marked as answer]
You don't provide any information about the shaders you are using or what you switched from so I can only guess what might be relevant.

If you are using the built-in ShaderSet then perhaps the change to require VSG_ALPHA_TEST (equivilant to GL_ALPHA_TEST in OpenGL) for enabling the alpha test in the shaders.  I wrote about the change on this post: #1168
